snappiershot/snapshot/metadata.py

```diff
@@ -34,6 +37,57 @@ def as_dict(self) -> Dict:
         dct["arguments"] = dct.pop("caller_info").args
         return dct
 
+    def compare_instantiated_objects(
+        self, obj_from_metadata: Any, obj_from_snapshot_file: Any
+    ) -> bool:
+        """ Instantiate object from a dictionary read from file in order to compare to metadata.
+
+        While encoding complex objects, an object decomposed into its parts in the form of a dictionary is written
+        to the snapshot JSON. Thus, snappiershot must attempt to re-instantiate those objects to do a direct
+        comparison.
+
+        Args:
+            obj_from_metadata: An object coming from the metadata.
+            obj_from_snapshot_file: An object (or decomposed object in the form of a dictionary) read out of the
+            snapshot file.
+        """
+        # Determine if the metadata object can instantiate a class from a dictionary, or whether any object is None
+        metadata_obj_has_method = hasattr(obj_from_metadata, "from_dict")
+        metadata_obj_is_not_none = obj_from_metadata is not None
+        obj_from_file_is_not_none = obj_from_snapshot_file is not None
+
+        # If object has special coding defined to skip checking, set a flag
+        metadata_obj_can_be_skipped = hasattr(
+            obj_from_metadata, SPECIAL_ENCODING_FUNCTION_NAME
+        )
+
+        result = False  # initialize result to false
+
+        if isinstance(obj_from_metadata, ndarray) or isinstance(
```

Comment:
I want it to do `all(a == b)` for the following cases:

- both are numpy array
- one is numpy array and the other is just a numeric list

Because if either (or both) are numpy array then it returns an array of booleans. Is there a better way to write this or should I just be explicit and write out all the conditionals.

Reply:
Ah ok this makes sense. Just explain that in a comment: "if at least one is a numpy array, elementwise comparison can be done like this:"

Reply:
[comment added ](https://github.com/MORSECorp/snappiershot/pull/96/commits/989acb9dc304ba0dbbc5ee5d304c2164b1020377)